Panic in executor plugins cache.FilteringResourceEventHandler

## Summary

What happened/what you expected to happen?
controller panic occasionally happend when configmaps in ns are deleted.
What version are you running?
v3.3.0

## Diagnostics
```
E0429 09:59:40.536415      34 runtime.go:78] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(nil), concrete:(*runtime._type)(0x1c76ae0), asserted:(*runtime._type)(0x1dc6060), missingMethod:"GetAnnotations"} (interface conversion: cache.DeletedFinalStateUnknown is not v1.Object: missing method GetAnnotations)
goroutine 225 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1be4fc0, 0xc059d4f680})
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/runtime/runtime.go:74 +0x85
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0})
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/runtime/runtime.go:48 +0x75
panic({0x1be4fc0, 0xc059d4f680})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/argoproj/argo-workflows/v3/workflow/controller.(*WorkflowController).newConfigMapInformer.func2({0x1c76ae0, 0xc033aed000})
        /go/src/github.com/argoproj/argo-workflows/workflow/controller/controller.go:1084 +0x34
k8s.io/client-go/tools/cache.FilteringResourceEventHandler.OnDelete({0x1f5c3a0, {0x21532f0, 0xc000b0c288}}, {0x1c76ae0, 0xc033aed000})
        /go/pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/controller.go:285 +0x46
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /go/pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/shared_informer.go:779 +0xdf
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f44157445d8)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000198f38, {0x211fba0, 0xc02a0a0000}, 0x1, 0xc02a09e000)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00028ed20, 0x3b9aca00, 0x0, 0x0, 0xc000198fb0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc0009aea00)
        /go/pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/shared_informer.go:771 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:73 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:71 +0x88
panic: interface conversion: cache.DeletedFinalStateUnknown is not v1.Object: missing method GetAnnotations [recovered]
        panic: interface conversion: cache.DeletedFinalStateUnknown is not v1.Object: missing method GetAnnotations

goroutine 225 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x0})
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/runtime/runtime.go:55 +0xd8
panic({0x1be4fc0, 0xc059d4f680})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/argoproj/argo-workflows/v3/workflow/controller.(*WorkflowController).newConfigMapInformer.func2({0x1c76ae0, 0xc033aed000})
        /go/src/github.com/argoproj/argo-workflows/workflow/controller/controller.go:1084 +0x34
k8s.io/client-go/tools/cache.FilteringResourceEventHandler.OnDelete({0x1f5c3a0, {0x21532f0, 0xc000b0c288}}, {0x1c76ae0, 0xc033aed000})
        /go/pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/controller.go:285 +0x46
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /go/pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/shared_informer.go:779 +0xdf
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x7f44157445d8)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:155 +0x67
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000198f38, {0x211fba0, 0xc02a0a0000}, 0x1, 0xc02a09e000)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:156 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00028ed20, 0x3b9aca00, 0x0, 0x0, 0xc000198fb0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:133 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc0009aea00)
        /go/pkg/mod/k8s.io/client-go@v0.21.5/tools/cache/shared_informer.go:771 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:73 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /go/pkg/mod/k8s.io/apimachinery@v0.21.5/pkg/util/wait/wait.go:71 +0x88
```

ResourceEventHandler may recieve DeletedFinalStateUnknown in some cases.
DeletedFinalStateUnknown cannot cast to v1.Object.

---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

--- a/workflow/controller/controller.go
+++ b/workflow/controller/controller.go
@@ -14,6 +14,7 @@ import (
 	"golang.org/x/time/rate"
 	apiv1 "k8s.io/api/core/v1"
 	apierr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/labels"
@@ -1020,7 +1021,10 @@ func (wfc *WorkflowController) newConfigMapInformer() cache.SharedIndexInformer 
 	if wfc.executorPlugins != nil {
 		indexInformer.AddEventHandler(cache.FilteringResourceEventHandler{
 			FilterFunc: func(obj interface{}) bool {
-				cm := obj.(metav1.Object)
+				cm, err := meta.Accessor(obj)
+				if err != nil {
+					return false
+				}
 				return cm.GetLabels()[common.LabelKeyConfigMapType] == common.LabelValueTypeConfigMapExecutorPlugin
 			},
 			Handler: cache.ResourceEventHandlerFuncs{

